### PR TITLE
test: fix flaky test `Send ETH from inside MetaMask finds the transaction`

### LINKED ISSUE #24574: Flaky Test: "Send ETH from inside MetaMask finds the transaction in the transactions list when sending to a Multisig Address"?
## Failure

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/80402/workflows/41cbd58d-f379-4fb3-98a3-56004e8b8872/jobs/2836380/tests

## Screenrecording 

https://github.com/MetaMask/metamask-extension/assets/54408225/9e3c7e0c-ac7c-48a9-b147-7faf3c1b8c7f

Gas Limit is set to 0 and there is no way the gas values are updated, even after subsequent successful gas api calls  (mocked responses)

Screenshot from 2024-05-20 15-41-32

### PR DESCRIPTION
**Pull request by seaona**
## **Description**
The problem with this flaky test is that in some occurrences we are proceeding to the last Confirmation screen without the balance being loaded (race condition). This can be observed at second 4 from this video. This test fails every time this condition is met (very often in Mutlichain build, possibly due to shorter flow of UI?)
The fix is simply waiting for the balance to be loaded before proceeding with the actions.

However, I think we should also revise the application behaviour, as it does not seem correct that:
- whenever we proceed to the last screen without the balance being loaded, the gas is set to 0 (specifically the gas limit)
- then, no matter how long we wait, the gas cannot be recovered from this and we are stucked in the confirmation screen (Confirm is disabled) --> this happens despite the balance being then loaded and also getting numerous successful responses from the gas API

I'll open separate tickets for that

Example of failure: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/81147/workflows/ec5f9950-23fa-4f6f-b248-9f894813e75d/jobs/2874889/tests

https://github.com/MetaMask/metamask-extension/assets/54408225/93b9bd5d-a9d2-4898-b743-8a59eed0eea2

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24639?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24574

## **Manual testing steps**

1. Check ci jobs
2. Run test multiple times locally  `yarn test:e2e:single test/e2e/tests/transaction/send-eth.spec.js --browser=chrome --leave-running=true --retry-until-failure --retries=15`

## **Screenshots/Recordings**
In the test failing artifacts we can see how we cannot proceed with the transaction as the Confirm button is disabled. However the problem is that the gas is set to 0 and never recovered, but the real root cause comes from proceeding with the previous screen without the balance loaded.

image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by zone-live — APPROVED** [review]

File: test/e2e/tests/transaction/send-eth.spec.js [hunk 1]
```diff
@@ -161,11 +161,11 @@ describe('Send ETH', function () {
           smartContract,
           title: this.test.fullTitle(),
         },
-        async ({ driver, contractRegistry }) => {
+        async ({ driver, contractRegistry, ganacheServer }) => {
           const contractAddress = await contractRegistry.getContractAddress(
             smartContract,
           );
-          await unlockWallet(driver);
+          await logInWithBalanceValidation(driver, ganacheServer);
```
Comment (chloeYue) [inline test/e2e/tests/transaction/send-eth.spec.js:168]: Hi 👋 , could you explain more why this change solves the flakiness? Thanks !

Comment (seaona) [reply test/e2e/tests/transaction/send-eth.spec.js:168]: hey :wave: basically this method will make sure that the balance is loaded before we proceed with any action, which is precisely the race condition that happens when this error occurs: we try to perform a tx's when the balance is still not loaded (see vid in the description).
This causes not being able to proceed with the Confirm button, and sets gas to 0, without a way of recovering from this state.

See snippet:
```
  const balance = await ganacheServer.getBalance();
  await driver.findElement({
    css: '[data-testid="eth-overview__primary-currency"]',
    text: `${balance} ETH`,
  });
```

Let me know if this clarifies :pray:

Comment (chloeYue) [reply test/e2e/tests/transaction/send-eth.spec.js:168]: Gotcha, thanks !

**Review by chloeYue — APPROVED** [review]
LGTM
